.github/workflows/merge-commits-to-next-branch.yml

```diff
@@ -29,72 +29,73 @@ jobs:
             const queueClient = queueServiceClient.getQueueClient(queueName);
             var receivedMessages = await queueClient.receiveMessages();
             const message = receivedMessages.receivedMessageItems[0];
-            $COMMIT_INFO = JSON.parse(message.messageText);
+            const firstMessage = JSON.parse(message.messageText);
+            return firstMessage;
+    outputs:
+      LABEL: ${{ fromJSON(steps.deque-message.outputs.result).label }}
+      AUTHOR: ${{ fromJSON(steps.deque-message.outputs.result).author }}
+      SHA: ${{ fromJSON(steps.deque-message.outputs.result).sha }}
   create-branch:
     runs-on: ubuntu-latest
-    if: ( ($COMMIT_INFO).label == 'queued' )
+    permissions:
+      contents: write
+      packages: write
+    needs: [ dequeue ]
+    if: (needs.dequeue.result) == 'success'
     steps:
-      - name: Create branch
-        uses: actions/github-script@7a5c598405937d486b0331594b5da2b14db670da # pin@v6
-        with:
-          script: |
-            git fetch origin
-            git checkout $TARGET_BRANCH
-            git checkout -b main-next-($COMMIT_INFO).sha
-            git merge ($COMMIT_INFO).sha --no-ff
-  reset-branch:
-    runs-on: ubuntu-latest
-    if: ( $COMMIT_INFO.label == 'queued' && needs.create-branch.result == 'failure'
-      )
-    steps:
-      - name: Reset main to a commit
-        uses: actions/github-script@7a5c598405937d486b0331594b5da2b14db670da # pin@v6
+      - run: echo ${{needs.dequeue.outputs.LABEL}}
+      - name: Git checkout
+        if: ${{needs.dequeue.outputs.LABEL}} == 'queued'
+        uses: actions/checkout@7884fcad6b5d53d10323aee724dc68d8b9096a2e # pin@v2
         with:
-          script: |
-            git fetch origin
-            git checkout $SOURCE_BRANCH
-            git branch -D main-next-($COMMIT_INFO).sha
-            git checkout -b main-next-($COMMIT_INFO).sha
-            git reset --hard ($COMMIT_INFO).sha
+          token: ${{ secrets.WORKFLOW_TOKEN }}
+          fetch-depth: '0'
+      - name: Create branch
+        id: create
+        if: ${{needs.dequeue.outputs.LABEL}} == 'queued'
+        run: |
+          git config user.name "Automation Bot"
+          git config user.email "<>"
+          git fetch origin
+          git checkout $TARGET
+          git checkout -b main-next-${{needs.dequeue.outputs.SHA}}
+          git fetch --all
+          git merge ${{needs.dequeue.outputs.SHA}} --no-ff
+          git push --set-upstream origin main-next-${{needs.dequeue.outputs.SHA}}
   createPullRequest:
     runs-on: ubuntu-latest
-    needs: [ create-branch ]
-    if: ( ($COMMIT_INFO).label == 'queued' && needs.create-branch.result == 'success'
-      )
+    needs: [ create-branch, dequeue ]
+    if: needs.create-branch.result == 'success'
     steps:
-      - name: Create Pull Request
+      - run: npm install @octokit/core
```

Comment:
can you use a global install? This is basically installing the package as a dependency of the main repo, which will be slow and unnecessary.

Reply:
testing this as part of a follow-up PR